main-app/MoveMaster-app/Assets/Scripts/Scoreboard/ScoreboardManager.cs

```diff
@@ -0,0 +1,43 @@
+using System.Collections;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+
+public class ScoreboardManager : MonoBehaviour
+{
+    public static ScoreboardManager Instance;
+    [SerializeField]
+    private TMP_Text scorePoints;
+    [SerializeField]
+    private int score=0;
+
+    void Awake(){
+        if (Instance != null && Instance != this) 
+    { 
+        Destroy(this); 
+    } 
+    else 
+    { 
+        Instance = this; 
+    } 
+    }
+
+    void Start()
+    {
+        setScore(0);
+    }
+
+    public void setScore(int newScore){
+        score = newScore;
+        scorePoints.SetText(newScore.ToString());
+    }
+
+    public int getScore(){
+        return score;
+    }
+
+    public void addValue(int value){
+        int newScore = score + value;
+        setScore(newScore);
+    }
```

Comment:
Wydaje mi się że dwie funkcje do ustawiania punktacji to przerost formy nad treścią, można zrobić jedną i po prostu score += value
scorePoints.SetText(score.ToString());

Reply:
Tu się nie zgadzam

Można do jednej linijki nawet pierdolnąć, ale nie uważam, że mniej znaczy lepiej w tym przypadku.
Dobrym nawykiem jest czasami wprowadzanie zmiennych pomocniczych, ponieważ poprawia to czytelność kodu, natomiast w tym przypadku może się to wydawać zbędne

Reply:
Nie do poprawy